webhooks-extension/config/openshift-development/openshift-tekton-webhooks-extension-development.yaml

```diff
@@ -0,0 +1,697 @@
+- apiGroups:
+  - tekton.dev
+  resources:
+  - taskruns/finalizers
+  - pipelineruns/finalizers
+  verbs:
+  - get
+  - list
+  - create
+  - update
+  - delete
+  - patch
+  - watch
+- apiGroups:
+  - tekton.dev
+  resources:
+  - tasks/status
+  - clustertasks/status
+  - taskruns/status
+  - pipelines/status
+  - pipelineruns/status
+  verbs:
+  - get
+  - list
+  - create
+  - update
+  - delete
+  - patch
+  - watch
+- apiGroups:
+  - extensions
+  resources:
+  - ingresses
+  - ingresses/status
+  verbs:
+  - delete
+  - create
+  - patch
+  - get
+  - list
+  - update
+  - watch
+- apiGroups:
+  - route.openshift.io
+  resources:
+  -  routes
+  verbs:
+  - get
+  - list
+  - create
+  - delete
+  - update
+  - patch
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: tekton-webhooks-extension-minimal
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: tekton-webhooks-extension-minimal
+subjects:
+- kind: ServiceAccount
+  name: tekton-webhooks-extension
+  namespace: tekton-pipelines
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels:
+    app: tekton-webhooks-extension
+  name: tekton-webhooks-extension-eventlistener
+  namespace: tekton-pipelines
+---
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: tekton-triggers-minimal
+rules:
+- apiGroups:
+  - tekton.dev
+  resources:
+  - tasks
+  - taskruns
+  - eventlisteners
+  - triggerbindings
+  - triggertemplates
+  verbs:
+  - get
+- apiGroups:
+  - tekton.dev
+  resources:
+  - pipelineruns
+  - pipelineresources
+  - taskruns
+  verbs:
+  - create
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: tekton-webhooks-extension-eventlistener-minimal
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: tekton-triggers-minimal
+subjects:
+- kind: ServiceAccount
+  name: tekton-webhooks-extension-eventlistener
+  namespace: tekton-pipelines
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: webhooks-extension
+  name: webhooks-extension
+  namespace: tekton-pipelines
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: webhooks-extension
+  template:
+    metadata:
+      labels:
+        app: webhooks-extension
+    spec:
+      containers:
+      - env:
+        - name: PORT
+          value: "8080"
+        - name: INSTALLED_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        - name: DOCKER_REGISTRY_LOCATION
+          value: DOCKER_REPO
+        - name: WEB_RESOURCES_DIR
+          value: web
+        # If this endpoint's protocol is https, ssl verification will be enabled on the github webhook
+        # openshift_master_default_subdomain usually of the format 'apps.host.company.com'
+        - name: WEBHOOK_CALLBACK_URL
+          value: "http://el-tekton-webhooks-eventlistener-tekton-pipelines.apps.motion.os.fyre.ibm.com"
```

Comment:
Do we need this `WEBHOOK_CALLBACK_URL`? We're the ones setting up the Ingress/Route, so can't we get the host from the Route?

Reply:
- For the ingress one, we use the input in the creation of the ingress.
- For the route, openshift takes care of it and the callback URL is simply used in creation of the actual webhook.

One argument for keeping the variable (other than I don't know how we know what values to use in the ingress case) ... would be that it could give users the ability to specify an alternative endpoint for their GitHub webhook to talk to.  Could it be the case where the "route" is not exposed, and the GitHub webhook has to talk to the route via some other "name/endpoint" which routes onwards to the Route?

I am open to the opinions of others here @a-roberts @mnuttall ??